File: picture/src/main/java/cn/zgy/picture/PictureUtils.java
package cn.zgy.picture;

import android.graphics.Bitmap;

public class PictureUtils {

    static {
        System.loadLibrary("picture-edit");
    }

    public static native String stringFromJNI();

    /**
     * 美白
     */
    public static native void ndkMb(Bitmap bitmap, float brightness, float constraint);

    /**
     * 灰度
     */
    public static native void grayScale(Bitmap bitmap);

    /**
     * 高斯模糊 1:成功
     */
    public static native int gaussBlur(Bitmap bitmap);
}

File: picture/src/main/java/cn/zgy/picture/PictureUtils.kt
package cn.zgy.picture

import android.graphics.Bitmap

class PictureUtils {

    companion object {
        init {
            System.loadLibrary("picture-edit")
            System.loadLibrary("compress")
        }
    }

    external fun stringFromJNI() : String

    /**
     * 美白
     */
    external fun ndkMb(bitmap : Bitmap, britness : Float, constrait : Float)

    /**
     * 灰度
     */
    external fun grayScale(bitmap : Bitmap)

    /**
     * 高斯模糊
     */
    external fun gaussBlur(bitmap: Bitmap)

    /**
     * 压缩
     */
    external fun compressBitmap(bitmap: Bitmap, quality: Int, filenameBytes: ByteArray, optimize: Boolean) : String
}